File: Transact.Data/Models/UserAccount.cs
using System;
using System.Collections.Generic;

#nullable disable

namespace Transact.Data.Models
{
    public partial class UserAccount
    {
        public UserAccount()
        {
            InverseCreatedByNavigation = new HashSet<UserAccount>();
            TenantAccesses = new HashSet<TenantAccess>();
        }

        public Guid Id { get; set; }
        public string LastName { get; set; }
        public string FirstName { get; set; }
        public string MiddleName { get; set; }
        public DateTime? CreatedOn { get; set; }
        public Guid? CreatedBy { get; set; }
        public Guid? RoleId { get; set; }
        public string Email { get; set; }

        public virtual UserAccount CreatedByNavigation { get; set; }
        public virtual Role Role { get; set; }
        public virtual ICollection<UserAccount> InverseCreatedByNavigation { get; set; }
        public virtual ICollection<TenantAccess> TenantAccesses { get; set; }
    }
}

File: Transact.Web/src/app/administration/useraccounts/useraccount.ts
import { Tenant } from '../tenants/tenant';

export class UserAccount {
  public Id: string | null = null;
  public LastName: string | null = null;
  public FirstName: string | null = null;
  public MiddleName: string | null = null;
  public RoleId: string | null = null;
  public Email: string | null = null;
  public Tenants?: Tenant[] = [];
}
export class User {
  public id: string | null = null;
  public lastName: string | null = null;
  public firstName: string | null = null;
  public password: string | null = null;
  public middleName: string | null = null;
  public roleId: number | null = null;
  public email: string | null = null;
}

export interface Role {
  Id: string;
  Name: string;
}
